### PR TITLE
testcase/kernel : Unlock mutex before pthread_mutex_destroy

### PR DESCRIPTION
**Pull request by hasw7569**
If g_cond_sig_val is not same with VAL_ONE run pthread_mutex_destroy.
So I added that unlock g_mutex before pthread_mutex_destory.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [69a43f1800be0b27e5743d375cf667ba78893fe9] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [69a43f1800be0b27e5743d375cf667ba78893fe9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7219239fece428612697945f111e25c29d09ac5d] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [7219239fece428612697945f111e25c29d09ac5d] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_pthread.c [hunk 1]
```diff
@@ -1290,29 +1290,37 @@ static void tc_pthread_pthread_cond_signal_wait(void)
...
+	TC_ASSERT_EQ_CLEANUP("pthread_cond_signal", g_cond_sig_val, VAL_THREE, goto errout);
 
 	pthread_cancel(pthread_waiter);
 	pthread_join(pthread_waiter, NULL);
 
+	ret_chk = pthread_mutex_destroy(&g_mutex);
+	TC_ASSERT_EQ("pthread_mutex_destroy", ret_chk, OK);
+
 	TC_SUCCESS_RESULT();
+	return;
```
Comment (jeongchanKim) [inline apps/examples/testcase/le_tc/kernel/tc_pthread.c]: This is not needed.

**Comment by seinfra** [comment]
Target : [557b3b367188a15d71c00cb82781092e7f4e84ab] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [557b3b367188a15d71c00cb82781092e7f4e84ab] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5fc65dddbc15b56b1282e7570d1611c390948fef] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [5fc65dddbc15b56b1282e7570d1611c390948fef] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [029bccc25e26d21f57d70fa9e389ce56701f4f81] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [029bccc25e26d21f57d70fa9e389ce56701f4f81] - Code Rule Check OK.

**Pull request merged**